diff --git a/src/main/java/io/hexlet/project_devops_deploy/storage/StorageConfig.java b/src/main/java/io/hexlet/project_devops_deploy/storage/StorageConfig.java
--- a/src/main/java/io/hexlet/project_devops_deploy/storage/StorageConfig.java
+++ b/src/main/java/io/hexlet/project_devops_deploy/storage/StorageConfig.java
@@ -4,6 +4,7 @@ import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
 
 @Configuration
 @EnableConfigurationProperties(S3StorageProperties.class)
@@ -12,8 +13,10 @@ public class StorageConfig {
     @Bean
     @ConditionalOnMissingBean(ImageStorageService.class)
     public ImageStorageService imageStorageService(S3StorageProperties properties) {
-        if (properties.bucket() != null && properties.region() != null && properties.accessKey() != null
-                && properties.secretKey() != null) {
+        if (StringUtils.hasText(properties.bucket())
+                && StringUtils.hasText(properties.region())
+                && StringUtils.hasText(properties.accessKey())
+                && StringUtils.hasText(properties.secretKey())) {
             return new S3ImageStorageService(properties);
         }
         return new LocalImageStorageService();
